kore/test/Test/Kore/Builtin/Map.hs

```diff
@@ -1582,7 +1582,6 @@ distinctVariables variables =
 -- | Utility function for mapping over the keys of a 'HashMap'.
 mapKeys ::
     Hashable k2 =>
-    Eq k2 =>
     (k1 -> k2) ->
     HashMap k1 v ->
     HashMap k2 v
```

Comment:
Just stumbled over this... (and likewise [`mapKeysAndValues` earlier](https://github.com/runtimeverification/haskell-backend/pull/3461/files#diff-030692d92b2a70d51d900894835f4799ecfd91017914f5bacf8067cd18f4d838L445)). Please create a ticket to investigate how we can get rid of it.
`unordered-containers` has its own [`mapKeys` function, with a big warning sign on it for non-injective key mappings](https://hackage.haskell.org/package/unordered-containers-0.2.19.1/docs/Data-HashMap-Internal.html#v:mapKeys). Every call site for this function should ensure the mapping is injective.

Reply:
I can open a ticket, but figuring out what's actually getting mapped where is going to be pretty hairy, going through all sorts of things like `Injection` instances. `mapKeysAndValues` seems like a higher priority, since it's in library code rather than test code. It's not exactly obvious that the way it's used is safe in the face of the renaming that happens within `mapVariables`. Maybe?

Reply:
Ticket #3476 opened.